src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -504,6 +505,17 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                         }
                         searchPattern = searchPattern.Substring(lastSlash + 1);
                     }
+
+                    // Typically we shouldn't see either of these cases
+                    if (searchPattern.IndexOfAny(s_EscapeCharacters) != -1)
```

Comment:
Does this help with throughput?  Replace shouldn't actually allocate a new string unless there's something to replace.